src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -8572,11 +8572,37 @@ def wrapped_func(*args, **kwargs):  # type: ignore[no-untyped-def] # pragma: no
     def _apply_snowpark_python_function_to_columns(
         self,
         snowpark_function: Callable,
+        kwargs: Any,  # possible named arguments which need to be added
     ) -> "SnowflakeQueryCompiler":
         """Apply Snowpark Python function to columns."""
 
         def sf_function(col: SnowparkColumn) -> SnowparkColumn:
-            return snowpark_function(col)
+            if not kwargs:
+                return snowpark_function(col)
+            # we have named kwargs, which may be positional
+            # in nature, and we need to align them to the snowpark
+            # function call alongside the column reference
+            # Get the total arg count for the function
+            function_arg_count = snowpark_function.__code__.co_argcount
+            # Get all variables for the function and slice off only the arguments
+            positional_args = snowpark_function.__code__.co_varnames[
+                :function_arg_count
+            ]
+            resolved_positional = []
+            col_specified = False
+            for arg in positional_args:
+                if arg in kwargs:
+                    resolved_positional.append(kwargs[arg])
+                else:
+                    if not col_specified:
+                        resolved_positional.append(col)
+                        col_specified = True
+                    else:
+                        ErrorMessage.not_implemented(
+                            f"Unspecified Argument: {arg} - when using apply with kwargs, all function arguments should be specified except the single column reference (if applicable)."
```

Comment:
will this message will be raised if the function takes multiple columns? should we make clearer that this isn't yet supported?

Reply:
So this should *only* be hit on the snowpark functions we explicitly allow - and... yeah; say you have a function like:

`func(A=<scalar>, B=<col>, C=<col>)` 

and you pass a kwargs is { A: "blabla" } then the code would assume that B references "self" and then it would throw an error when it saw another parameter "C"

To be honest; if we didn't already limit these functions to snowpark ones I think I would try to articulate a different path - but supporting apply with functions which have multiple column references is a little outside the bounds of pandas too.